src/Data/Types/URI.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace LORIS\Data\Types;
+
+/**
+ * A URI represents a data type that represents a resource by
+ * storing a pointer to where the resource is located.
+ *
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ */
+class URI implements \LORIS\Data\Type
+{
+    /**
+     * Convert data type to a string
+     *
+     * @return string
+     */
+    public function __toString()
+    {
+        return "URI";
+    }
+
+    /**
+     * Serialize to JSON by converting to a string.
+     *
+     * @return string
+     */
+    public function jsonSerialize()
+    {
+        return $this->__toString();
+    }
+
+    /**
+     * URIs are stored in the SQL database as varchar columns.
+     *
+     * @return string
+     */
+    public function asSQLType() : string
+    {
+        return "varchar(255)";
```

Comment:
This can be tight. What about text?

Reply:
text seems excessive, what do we do for existing URLs in the database? I based this on the `File` column in the files table.

Reply:
We usually construct them using chunks from the database (e.g: basename, host, ...).
I tested our database class and it throws an exception when mysql emits a warning (data too long in this case.) But I am not sure if the mri_pipeline perl script does the same.

If we use text instead of varchar(255), it will cost 1 additional byte per row.  I don't think the cost is high enough to counterweight the flexibility is gives.

Reply:
Sorry for the delay responding.. my concern isn't that it's too heavy or light, it's that it doesn't match what our columns in the database currently are. Regardless, I changed it to "text" since this isn't actually used anywhere and the dictionary items that will end up using it will probably end up being constructed, as you say..